src/unix/fs.c

```diff
@@ -1082,7 +1082,8 @@ int uv_fs_read(uv_loop_t* loop, uv_fs_t* req,
   if (req->bufs == NULL)
     return -ENOMEM;
 
-  memcpy(req->bufs, bufs, nbufs * sizeof(*bufs));
+  if (bufs)
```

Comment:
I think we should check if bufs was provided first thing and return UV_EINVAL if it's null, or nbufs is zero.